### PR TITLE
Extract the slow rendering detection instrumentation

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**

### CODE REVIEW

**Review by laurit — APPROVED** [review]

File: splunk-otel-android/src/main/java/io/opentelemetry/rum/internal/instrumentation/slowrendering/SlowRenderListener.java [hunk 1]
```diff
@@ -194,7 +189,7 @@ private void reportSlowRenders() {
         try {
             activities.forEach((activity, listener) -> reportSlow(listener));
         } catch (Exception e) {
-            Log.w(LOG_TAG, "Exception while processing frame metrics", e);
+            Log.w("OpenTelemetryRum", "Exception while processing frame metrics", e);
```
Comment (breedx-splk) [inline splunk-otel-android/src/main/java/io/opentelemetry/rum/internal/instrumentation/slowrendering/SlowRenderListener.java:192]: we should still make a constant field for that `OpenTelemetryRum` even if it's not `LOG_TAG`. What happened to that anyway?

File: splunk-otel-android/src/main/java/com/splunk/rum/RumInitializer.java [hunk 2]
```diff
@@ -191,15 +190,19 @@ SplunkRum initialize(
...
+                        SlowRenderingDetector.builder()
+                                .setSlowRenderingDetectionPollInterval(
+                                        builder.slowRenderingDetectionPollInterval)
+                                .build()
+                                .installOn(instrumentedApplication);
+                        initializationEvents.add(
+                                new RumInitializer.InitializationEvent(
+                                        "slowRenderingDetectorInitialized", timingClock.now()));
+                    });
+        }
```
Comment (breedx-splk) [inline splunk-otel-android/src/main/java/com/splunk/rum/RumInitializer.java:205]: If this was factored out to a named method like `initializeSlowRenderingDetection(otelRumBuilder)` or something, you could return early from the if and save some indent depth. It's getting a little burly in there.

**Review by breedx-splk — APPROVED** [review]
Looks good. 🤘🏻

File: splunk-otel-android/src/main/java/io/opentelemetry/rum/internal/instrumentation/slowrendering/SlowRenderListener.java (hunk 1 above)
Comment (mateuszrzeszutek) [reply splunk-otel-android/src/main/java/io/opentelemetry/rum/internal/instrumentation/slowrendering/SlowRenderListener.java:192]: `LOG_TAG` is placed in `SplunkRum`, can't use it here.

> we should still make a constant field for that `OpenTelemetryRum`

👍

File: splunk-otel-android/src/main/java/com/splunk/rum/RumInitializer.java (hunk 2 above)
Comment (mateuszrzeszutek) [reply splunk-otel-android/src/main/java/com/splunk/rum/RumInitializer.java]: Done. I also extracted similar private methods for other instrumentations.

**Pull request merged**